front/src/applications/stdcmV2/hooks/usePathfindingIsPossible.ts

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import { compact } from 'lodash';
+import { useSelector } from 'react-redux';
+
+import { osrdEditoastApi, type PathfindingResult } from 'common/api/osrdEditoastApi';
+import { useOsrdConfSelectors } from 'common/osrdContext';
+
+import { useStoreDataForRollingStockSelector } from 'modules/rollingStock/components/RollingStockSelector/useStoreDataForRollingStockSelector';
+
+import useInfraStatus from 'modules/pathfinding/hooks/useInfraStatus';
+import { getSupportedElectrification, isThermal } from 'modules/rollingStock/helpers/electric';
+import usePathProperties from 'modules/pathfinding/hooks/usePathProperties';
+
+const usePathfindingIsPossible = () => {
+  const { getPathSteps } = useOsrdConfSelectors();
+
+  const { infra } = useInfraStatus();
+  const pathSteps = useSelector(getPathSteps);
+  const { rollingStock } = useStoreDataForRollingStockSelector();
+
+  const [pathfinding, setPathfinding] = useState<PathfindingResult>();
+
+  const [postPathfindingBlocks] =
+    osrdEditoastApi.endpoints.postInfraByInfraIdPathfindingBlocks.useMutation();
+
+  const pathStepsLocation = useMemo(
+    () => pathSteps.map((step) => (step && (!('uic' in step) || step.uic !== -1) ? step : null)),
+    [pathSteps]
+  );
+
+  const pathProperties = usePathProperties(
+    infra?.id,
+    pathfinding?.status === 'success' ? pathfinding : undefined,
+    ['geometry']
+  );
+
+  useEffect(() => {
+    const launchPathfinding = async () => {
+      setPathfinding(undefined);
+      if (!infra || infra.state !== 'CACHED' || !rollingStock) {
```

Comment:
It might probably work like that too ?

```suggestion
      if (infra?.state !== 'CACHED' || !rollingStock) {
```

Reply:
Done